### PR TITLE
Improve a bit internal and public API

### PR DESCRIPTION
**Pull request by paugier**
I had to change few things to try to better understand some snek internals.

I propose to depreciate the function `source_root` and to rename it as `get_nek_source_root`.

### CODE REVIEW

File: src/snek5000/util/__init__.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def activate_paths():
     """Setup environment variables in preparation for Nek5000 build.
```
Comment (ashwinvis) [inline src/snek5000/util/__init__.py:112]: We can remove this function.

File: src/snek5000/output/base.py [hunk 2]
```diff
@@ -298,7 +290,7 @@ def __init__(self, sim=None, params=None):
         else:
             self.package = get_solver_package(self.name_solver)
 
-        self.root = self.get_root()
+        self.path_solver_package = self.get_path_solver_package()
```
Comment (ashwinvis) [inline src/snek5000/output/base.py:293]: Add a method `get_root`, which issues `DeprecationWarning` and returns `get_path_solver_package` instead. That will make merging this PR smoother.

**Review by ashwinvis — CHANGES_REQUESTED** [review]

File: src/snek5000/output/base.py (hunk 2 above)
Comment (paugier) [reply src/snek5000/output/base.py:308]: Done in a9cb307f7dc013407bc4e695ecd6d9cd690f1b33

File: src/snek5000/util/__init__.py (hunk 1 above)
Comment (paugier) [reply src/snek5000/util/__init__.py]: Done in fad6e499072ea3e69223554921dc611f4590866d

**Pull request merged**